### PR TITLE
[#6136] iexit: avoid removing auth file for service account (unless forced) (main)

### PR DESCRIPTION
**Pull request by SwooshyCueb**
Prerequisite: irods/irods#6969
Addresses irods/irods#6136 for main

Will backport to stable branches after reviews done.

### CODE REVIEW

File: src/iexit.cpp
Comment (trel) [inline src/iexit.cpp:40]: Do we want to wrap the `unlink` itself with `!isServiceAccount`?

That would prevent the environment file from being deleted too... 

I'm feeling a markdown table itching to get out...

Comment (SwooshyCueb) [reply src/iexit.cpp:40]: I don't quite understand the logic going on behind the scenes, but when `iexit` is run as the service account, it doesn't try to remove `irods_environment.json`, but `irods_environment.json.[SOME NUMBER]`

Comment (trel) [reply src/iexit.cpp:40]: oooh, right.   because there could be multiple connections/sessions for the same user.

carry on.

File: src/iexit.cpp
Comment (alanking) [inline src/iexit.cpp:38]: We should wrap this in a try-catch block for the `irods::exception`s.

File: src/iexit.cpp
Comment (alanking) [reply src/iexit.cpp:40]: Should the message on 42 come before the `unlink` on 40? I realize that would introduce 2 if blocks, just seems upside-down to me.

**Review by alanking — COMMENTED** [review]
Agreed on the point about parseCmdLineOpt. That function is evil.

Hopefully we won't need to worry about it once we can shift focus to irods_client_cli...

File: src/iexit.cpp
Comment (korydraughn) [inline src/iexit.cpp:19]: Is this required for the issue being addressed?

Comment (SwooshyCueb) [reply src/iexit.cpp:19]: No, it's more of a note for myself. I'm going to remove it once I've got an issue open for it.

File: src/iexit.cpp
Comment (alanking) [inline src/iexit.cpp:53]: Consider using `client_display_what` here to avoid printing the whole stack.

File: src/iexit.cpp
Comment (alanking) [inline src/iexit.cpp:7]: Since we'll be back here to remove the TODO below anyway, let's remove the Copyright lines and alphabetize these

**Review by alanking — COMMENTED** [review]
I'm assuming we are expecting one more change to remove the TODO once an issue is created, so I left a couple other notes. Looking real nice

**Review by alanking — APPROVED** [review]
Yep, seems good to me. Let's wait for one more approval and I think we can get this in.

File: src/iexit.cpp
Comment (korydraughn) [inline src/iexit.cpp:59]: Leftover? Perhaps this should be captured in an issue.

Comment (SwooshyCueb) [reply src/iexit.cpp:59]: #6986

**Review by korydraughn — COMMENTED** [review]
Once the review comment is resolved, please add the pound.

**Review by korydraughn — APPROVED** [review]

**Comment by SwooshyCueb** [comment]
Testing requires changing unix user. Not something we can feasibly do with our current test framework.

**Review by alanking — APPROVED** [review]

**Comment by korydraughn** [comment]
Cool cool.

**Pull request merged**
